torchvision/models/_utils.py

```diff
@@ -64,3 +67,127 @@ def forward(self, x):
+                return out
+            return self.create_proxy('call_module', module_qualified_name, args, kwargs)
+        finally:
+            self.current_module_qualified_name = old_qualname
+
+    def create_proxy(self, kind: str, target: torch.fx.node.Target, args: Tuple[Any, ...],
+                     kwargs: Dict[str, Any], name: Optional[str] = None, type_expr: Optional[Any] = None):
+        """
+        Override of `Tracer.create_proxy`. This override intercepts the recording
+        of every operation and stores away the current traced module's qualified
+        name in `node_to_originating_module`
+        """
+        proxy = super().create_proxy(kind, target, args, kwargs, name, type_expr)
+        self.node_to_originating_module[proxy.node] = self.current_module_qualified_name
+        return proxy
+
+
+def get_intermediate_layers(model: nn.Module, return_layers: Dict[str, str]) -> nn.Module:
+    """
+    Creates a new FX-based module that returns intermediate layers from a given model.
+    This is achieved by re-writing the computation graph of the model via FX to return
+    the requested layers.
+
+    All unused layers are removed, together with their corresponding parameters.
+
+    Args:
+        model (nn.Module): model on which we will extract the features
+        return_layers (Dict[name, new_name]): a dict containing the names
+            of the modules for which the activations will be returned as
+            the key of the dict, and the value of the dict is the name
+            of the returned activation (which the user can specify).
+
+    Examples::
+
+        >>> m = torchvision.models.resnet18(pretrained=True)
+        >>> # extract layer1 and layer3, giving as names `feat1` and feat2`
+        >>> new_m = torchvision.models._utils.get_intermediate_layers(m,
+        >>>     {'layer1': 'feat1', 'layer3': 'feat2'})
+        >>> out = new_m(torch.rand(1, 3, 224, 224))
+        >>> print([(k, v.shape) for k, v in out.items()])
+        >>>     [('feat1', torch.Size([1, 64, 56, 56])),
+        >>>      ('feat2', torch.Size([1, 256, 14, 14]))]
+    """
+    # TODO come up with a better name for this
+    # TODO have duplicate nodes but full coverage for module names
+    return_layers = {str(k): str(v) for k, v in return_layers.items()}
+
+    # Instantiate our ModulePathTracer and use that to trace the model
+    tracer = ModulePathTracer()
+    graph = tracer.trace(model)
+
+    name = model.__class__.__name__ if isinstance(model, nn.Module) else model.__name__
+    m = torch.fx.GraphModule(tracer.root, graph, name)
+
+    # check that all outputs in return_layers are present in the model
+    if not set(return_layers).issubset(tracer.node_to_originating_module.values()):
+        raise ValueError("return_layers are not present in model")
+
+    # Get output node
+    orig_output_node: Optional[torch.fx.Node] = None
+    for n in reversed(m.graph.nodes):
+        if n.op == "output":
```

Comment:
What happens in cases where we have multiple outputs (example `Inception3` which got auxiliaries)? It seems that FX has another node called `inception_outputs`:
```
>>> list(reversed(m.graph.nodes))
[output, inception_outputs, fc, flatten_1, dropout, ....]
```

You can see this by replacing this input on your test:
```
        model = torchvision.models.inception_v3(pretrained=False)
        return_layers = {'Mixed_7c': '0', 'avgpool': '1'}
```